apiserver/common/modelmanagerinterface.go

```diff
@@ -99,21 +101,13 @@ var _ ModelManagerBackend = (*modelManagerStateShim)(nil)
 
 type modelManagerStateShim struct {
 	*state.State
+	pool *state.StatePool
```

Comment:
Not sure i agree with this - the state shim is supposed to only wrap state to provide access to its methods.
I rather stick to keeping state and state pool separate in these shim objects to reflect the semantics of those structs